### ISSUE TITLE
Ability to filter livestreams with tags

### ISSUE THREAD

**Issue by Puncia** [bastimeyer/livestreamer-twitch-gui#219]
I don't know exactly how Twitch handles this, but they recently grouped a bunch of game sections under the Creative one. You can still watch those game sections by using tags (i.e. #programming) in the website.

However, if you now search for "Programming" you get like 1/10 of current livestreams (not sure what determines which stream to show as a result). Would be cool to have the ability to filter every channel under the Creative section; pretty sure they have the tag in their title, unless Twitch sorts it out differently.

**Comment by bastimeyer** [comment]
Whoever has implemented this on the Twitch website has done a **terrible** job IMO...
All the streams are being filtered on the client side after fetching everything from the `Creative` section. **Please read https://github.com/bastimeyer/livestreamer-twitch-gui/issues/151#issuecomment-183581175 why I think it's bad...**

The tags are just a simple list you can get from here:
https://api.twitch.tv/api/creative/tags

So when clicking a tag from this list, several API requests are being made and roughly **300+** streams are being fetched. Then all stream statuses are being filtered for the selected tag and every non-matching stream will be discarded.

The right approach would be filtering for stream statuses while doing the API request, like this:
https://api.twitch.tv/kraken/streams?limit=10&offset=0&game=Creative&status=programming
or
https://api.twitch.tv/kraken/streams?limit=10&offset=0&game=Creative&status=%23programming

Twitch doesn't support this, though... Don't ask me why, it's basically the same with filtering for multiple languages or filtering other stuff like games.

The only reason I'm seeing them not implementing or simply ignoring this could be the performance of their database and API. But when I see their website and all those **invalid** API requests being made by every viewer every couple of seconds, I just have to doubt this and I'm thinking that they simply don't care. It's a shame... (https://api.twitch.tv/kraken/streams?channel=static or https://api.twitch.tv/kraken/streams?channel=directory)

Long story short, I don't want to implement client side filtering.

**Comment by wyrde** [comment]
Are they still using this method for filtering? There have been a lot of changes made in creative the last year, I'd hope they would have made using the community hashes more efficient.

Though it does explain why filtering was so haphazard for so many months.
